src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,95 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int GROW_FACTOR = 2;
+
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[INITIAL_CAPACITY];
+        threshold = (int) (table.length * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = getIndexForKey(key);
+        Node<K, V> newNode = new Node<>(key, value, null);
+        if (size + 1 >= threshold) {
+            resize();
+        }
+        if (table[index] == null) {
+            table[index] = newNode;
+        } else {
+            Node<K, V> previousNode = table[index];
+            while (true) {
+                if (Objects.equals(previousNode.key, key)) {
```

Comment:
As far as I recall, it's not recommended to use utility classes like Objects.

Reply:
This time there isn't such restriction in `checklist.md`, but ok, I'll rewrite it

Reply:
it's fine now, feel free to use it from now on